src/coreclr/debug/daccess/request.cpp

```diff
@@ -3171,47 +3171,16 @@ ClrDataAccess::GetNestedExceptionData(CLRDATA_ADDRESS exception, CLRDATA_ADDRESS
 HRESULT
 ClrDataAccess::GetDomainLocalModuleData(CLRDATA_ADDRESS addr, struct DacpDomainLocalModuleData *pLocalModuleData)
 {
-    if (addr == 0 || pLocalModuleData == NULL)
-        return E_INVALIDARG;
-
-    SOSDacEnter();
-
-    DomainLocalModule* pLocalModule = PTR_DomainLocalModule(TO_TADDR(addr));
-
-    pLocalModuleData->pGCStaticDataStart    = TO_CDADDR(PTR_TO_TADDR(pLocalModule->GetPrecomputedGCStaticsBasePointer()));
-    pLocalModuleData->pNonGCStaticDataStart = TO_CDADDR(pLocalModule->GetPrecomputedNonGCStaticsBasePointer());
-    pLocalModuleData->pDynamicClassTable    = PTR_CDADDR(pLocalModule->m_pDynamicClassTable.Load());
-    pLocalModuleData->pClassData            = (TADDR) (PTR_HOST_MEMBER_TADDR(DomainLocalModule, pLocalModule, m_pDataBlob));
-
-    SOSDacLeave();
-    return hr;
+    // CoreCLR does not use domain local modules anymore
```

Comment:
Does this need to come with `SOS_BREAKING_CHANGE_VERSION` ?

Reply:
The check/warning message in SOS is only done for commands that assume runtime data structure layout like dumplog, threads -special, printexception. These 2 APIs are called in dumpobj which doesn't do the check and will fail with no warning.

It looks like SOS should do the check/warning on every command.  I didn't consider these type of API breaking changes just layout changes when I added the checks. I'll add a SOS tracking issue to address this.

In the end, yes, SOS_BREAKING_CHANGE_VERSION should be bumped.

Reply:
Interesting. I had interpreted the failures in !dumpobj as pretty reasonable and not breaking. The way it failed is fairly graceful, as instead of crashing, SOS would simply report "<no information>" for normal statics, and skip reporting for any TLS statics.

Reply:
> I'll add a SOS tracking issue to address this.

It impacts clrmd too.

> SOS would simply report "<no information>" for normal statics, and skip reporting for any TLS statics.

I think it is hard for a random person out there to connect the dots between SOS failing to dump the statics and needing a newer SOS version to fix that.